core/services/feeds/service.go

```diff
@@ -510,6 +510,21 @@ func (s *service) DeleteJob(ctx context.Context, args *DeleteJobArgs) (int64, er
 		logger.Errorw("Failed to push metrics for job proposal deletion", "err", err)
 	}
 
+	job, err := s.jobORM.FindJobByExternalJobID(ctx, proposal.ExternalJobID.UUID)
```

Comment:
`proposal.ExternalJobID` was created as `uuid.NullUUID`. Even if I don't think it will ever be nil I'd probably wrap the code above in between someting similar to here https://github.com/smartcontractkit/chainlink/blob/5774807c429984241dcb0aa30cbf842c3757469e/core/services/feeds/service.go#L1007
 
```
if proposal.ExternalJobID.Valid {

} else {
  error
}
```

Reply:
I don't think we necessarily need to error here. if jp.ExternalJobID isn't valid, it just means the job isn't in approved status. Proposals iff approved must have the externalJobID (I think there's a db check constraint on this)

Reply:
Actually I take that back. I meant to comment that this would happen if the job was not in approved status :D and to add a test for this. Otherwise deletion requests sent for proposals that were already cancelled or in some other non-approved status would error. I believe this will return a sql no row if not found. So you can skip those and error otherwise